bouquet/solver.py

```
@@ -431,6 +468,197 @@ def _perform_final_relaxation(
+
+    candidates = _select_ensemble_candidates(
+        state,
+        window_eV=window_eV,
+        p_threshold=ENSEMBLE_P_THRESHOLD,
+        sigma_floor_eV=sigma_floor_eV,
+        failure_energy_eV=FAILURE_ENERGY_EV,
+    )
+
+    # Tight, UNCONSTRAINED optimization of each candidate (no step limit).
+    optimized: List[Tuple[Atoms, float]] = []
+    for k, (coords, atoms) in enumerate(candidates):
```

Ruff (0.15.14)

[warning] Loop control variable `coords` not used within loop body

Rename unused `coords` to `_coords`

(B007)